packages/rum-core/src/domain/view/viewMetrics/trackScrollMetrics.spec.ts

```diff
@@ -26,117 +59,61 @@ describe('trackScrollMetrics', () => {
-    let configuration: RumConfiguration
-
-    const getMetrics = jasmine.createSpy('getMetrics')
-
-    const newScroll = (scrollParams: { scrollHeight: number; scrollDepth: number; scrollTop: number }) => {
-      getMetrics.and.returnValue(scrollParams)
-
-      window.dispatchEvent(createNewEvent(DOM_EVENT.SCROLL))
-
-      clock.tick(THROTTLE_SCROLL_DURATION)
-    }
-
-    beforeEach(() => {
-      configuration = {} as RumConfiguration
-      clock = mockClock()
-      stopTrackScrollMetrics = trackScrollMetrics(
-        configuration,
-        { relative: 0 as RelativeTime, timeStamp: 0 as TimeStamp },
-        (s) => (scrollMetrics = s),
-        getMetrics
-      ).stop
-    })
-
-    afterEach(() => {
-      stopTrackScrollMetrics()
-      scrollMetrics = undefined
-      clock.cleanup()
-    })
-
-    it('should update scroll metrics when scrolling the first time', () => {
-      newScroll({ scrollHeight: 1000, scrollDepth: 500, scrollTop: 100 })
-
-      expect(scrollMetrics).toEqual({
-        maxDepthScrollHeight: 1000,
-        maxDepth: 500,
-        maxDepthScrollTop: 100,
-        maxDepthTime: 1000 as Duration,
-      })
+  it('should update scroll height and scroll depth', () => {
+    scrollObservable.notify({ scrollDepth: 700, scrollHeight: 2000, scrollTop: 100 })
+    expect(callbackSpy).toHaveBeenCalledOnceWith({
+      maxDepth: 700,
+      maxDepthScrollHeight: 2000,
+      maxDepthTime: jasmine.any(Number),
+      maxDepthScrollTop: 100,
     })
-
-    it('should update scroll metrics when scroll depth has increased', () => {
-      newScroll({ scrollHeight: 1000, scrollDepth: 500, scrollTop: 100 })
-
-      newScroll({ scrollHeight: 1000, scrollDepth: 600, scrollTop: 200 })
-
-      expect(scrollMetrics).toEqual({
-        maxDepthScrollHeight: 1000,
-        maxDepth: 600,
-        maxDepthScrollTop: 200,
-        maxDepthTime: 2000 as Duration,
-      })
+  })
+  it('should update scroll height only if it has increased', () => {
+    scrollObservable.notify({ scrollDepth: 700, scrollHeight: 2000, scrollTop: 100 })
+    scrollObservable.notify({ scrollDepth: 700, scrollHeight: 1900, scrollTop: 100 })
+    expect(callbackSpy).toHaveBeenCalledOnceWith({
+      maxDepth: 700,
+      maxDepthScrollHeight: 2000,
+      maxDepthTime: jasmine.any(Number),
+      maxDepthScrollTop: 100,
     })
-
-    it('should NOT update scroll metrics when scroll depth has not increased', () => {
-      newScroll({ scrollHeight: 1000, scrollDepth: 600, scrollTop: 200 })
-
-      newScroll({ scrollHeight: 1000, scrollDepth: 450, scrollTop: 50 })
-
-      expect(scrollMetrics).toEqual({
-        maxDepthScrollHeight: 1000,
-        maxDepth: 600,
-        maxDepthScrollTop: 200,
-        maxDepthTime: 1000 as Duration,
-      })
+    expect(callbackSpy).not.toHaveBeenCalledOnceWith({
+      maxDepth: 700,
+      maxDepthScrollHeight: 1900,
+      maxDepthTime: jasmine.any(Number),
+      maxDepthScrollTop: 100,
     })
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏remove this assertion. It is useless, since `toHaveBeenCalledOnceWith` already ensures that it has been called only once with the provided argument.